### PR TITLE
[receiver/hostmetrics] Add `process.signals_pending` metric

### PR DESCRIPTION
**Pull request by andrzej-stencel**
**Description:**

This pull request adds a new optional metric named `process.signals_pending` to the metrics scraped by the `process` scraper of the `hostmetrics` receiver.

Adding this metric to the semantic conventions has been proposed here:
- https://github.com/open-telemetry/opentelemetry-specification/issues/2826.

**Link to tracking Issue:** 

- https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/14084

### CODE REVIEW

**Comment by dmitryax** [comment]
please rebase

File: receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go [hunk 1]
```diff
@@ -229,6 +235,25 @@ func (s *scraper) scrapeAndAppendDiskIOMetric(now pcommon.Timestamp, handle proc
...
+	}
+
+	rlimitStats, err := handle.RlimitUsage(true)
+	if err != nil {
+		return err
+	}
+
+	for _, rlimitStat := range rlimitStats {
+		if rlimitStat.Resource == process.RLIMIT_SIGPENDING {
+			s.mb.RecordProcessSignalsPendingDataPoint(now, int64(rlimitStat.Used))
```
Comment (dmitryax) [inline receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go:250]: Should we add `break` statement here?

Comment (astencel-sumo) [reply receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go:250]: Good point, no reason to continue the loop.

**Comment by astencel-sumo** [comment]
Rebased and squashed. @dmitryax, @dashpole can you review?

File: receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go [hunk 2]
```diff
@@ -132,6 +134,10 @@ func (s *scraper) scrape(_ context.Context) (pmetric.Metrics, error) {
 			errs.AddPartial(fileDescriptorMetricsLen, fmt.Errorf("error reading open file descriptor count for process %q (pid %v): %w", md.executable.name, md.pid, err))
 		}
 
+		if err = s.scrapeAndAppendSignalsPendingMetric(now, md.handle); err != nil {
+			errs.AddPartial(diskMetricsLen, fmt.Errorf("error reading pending signals for process %q (pid %v): %w", md.executable.name, md.pid, err))
```
Comment (evan-bradley) [inline receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go:138]:
```suggestion
			errs.AddPartial(signalMetricsLen, fmt.Errorf("error reading pending signals for process %q (pid %v): %w", md.executable.name, md.pid, err))
```

File: receiver/hostmetricsreceiver/internal/scraper/processscraper/process.go [hunk 3]
```diff
@@ -96,6 +96,7 @@ type processHandle interface {
 	PageFaults() (*process.PageFaultsStat, error)
 	NumCtxSwitches() (*process.NumCtxSwitchesStat, error)
 	NumFDs() (int32, error)
+	RlimitUsage(bool) ([]process.RlimitStat, error)
```
Comment (evan-bradley) [inline receiver/hostmetricsreceiver/internal/scraper/processscraper/process.go:99]: Can we name the boolean parameter to this method and explain what it is used for? Right now it's unclear what its effect is.

**Review by evan-bradley — COMMENTED** [review]
Mostly looks good to me. Given the apparent pause in adding metrics to the semantic conventions, what is the plan for adding this metric here?

File: receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go (hunk 2 above)
Comment (astencel-sumo) [reply receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go:138]: Oops I guess that's rebase magic doing its work. Thanks for noticing this Evan!

File: receiver/hostmetricsreceiver/internal/scraper/processscraper/process.go (hunk 3 above)
Comment (astencel-sumo) [reply receiver/hostmetricsreceiver/internal/scraper/processscraper/process.go]: Added parameter name and the comment from gopsutil describing the parameter.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**